scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -515,67 +535,198 @@ def dijkstra(csgraph, directed=True, indices=None,
 
     #------------------------------
     # initialize dist_matrix for output
-    dist_matrix = np.zeros((len(indices), N), dtype=DTYPE)
-    dist_matrix.fill(np.inf)
-    dist_matrix[np.arange(len(indices)), indices] = 0
+    if min_only:
+        dist_matrix = np.zeros((N), dtype=DTYPE)
+        dist_matrix.fill(np.inf)
+        dist_matrix[indices] = 0
+    else:
+        dist_matrix = np.zeros((len(indices), N), dtype=DTYPE)
+        dist_matrix.fill(np.inf)
+        dist_matrix[np.arange(len(indices)), indices] = 0
 
     #------------------------------
     # initialize predecessors for output
     if return_predecessors:
-        predecessor_matrix = np.empty((len(indices), N), dtype=ITYPE)
-        predecessor_matrix.fill(NULL_IDX)
+        if min_only:
+            predecessor_matrix = np.empty((N), dtype=ITYPE)
+            predecessor_matrix.fill(NULL_IDX)
+            source_matrix = np.empty((N), dtype=ITYPE)
+            source_matrix.fill(NULL_IDX)
+        else:
+            predecessor_matrix = np.empty((len(indices), N), dtype=ITYPE)
+            predecessor_matrix.fill(NULL_IDX)
     else:
-        predecessor_matrix = np.empty((0, N), dtype=ITYPE)
+        if min_only:
+            predecessor_matrix = np.empty((0), dtype=ITYPE)
+            source_matrix = np.empty((0), dtype=ITYPE)
```

Comment:
The extra parens aren't necessary. If you want it to be a one-element tuple, you'll need to add a comma: `(0,)`.

But I'd go with the plain `0`, personally.